src/components/widgets/NumberWidget.js

```diff
@@ -38,9 +38,6 @@ export default class NumberWidget extends React.Component {
     this.distance = 0;
     this.onMouseDownValue = 0;
     this.prevPointer = [0, 0];
-
-    this.setValue(this.props.value);
-    this.onBlur();
```

Comment:
This is the issue, truncating 5 decimals to 3 decimals and triggering the props.onChange callback.